src/random_lcg.cpp

```diff
@@ -28,66 +18,64 @@ constexpr uint64_t prn_stride {152917LL};                // stride between
                                                          //   particles
 constexpr double   prn_norm   {1.0 / prn_mod};           // 2^-63
 
-// Current PRNG state
-uint64_t prn_seed[N_STREAMS];  // current seed
-int      stream;               // current RNG stream
-#pragma omp threadprivate(prn_seed, stream)
-
-
 //==============================================================================
 // PRN
 //==============================================================================
 
-extern "C" double
-prn()
+double prn(uint64_t* prn_seed)
 {
   // This algorithm uses bit-masking to find the next integer(8) value to be
   // used to calculate the random number.
-  prn_seed[stream] = (prn_mult*prn_seed[stream] + prn_add) & prn_mask;
+  *prn_seed = (prn_mult * (*prn_seed) + prn_add) & prn_mask;
 
   // Once the integer is calculated, we just need to divide by 2**m,
   // represented here as multiplying by a pre-calculated factor
-  return prn_seed[stream] * prn_norm;
+  return (*prn_seed) * prn_norm;
 }
 
 //==============================================================================
 // FUTURE_PRN
 //==============================================================================
 
-extern "C" double
-future_prn(int64_t n)
+double future_prn(int64_t n, uint64_t prn_seed)
+{
+  return future_seed(static_cast<uint64_t>(n), prn_seed) * prn_norm;
+}
+
+//==============================================================================
+// INIT_SEED
+//==============================================================================
+
+void init_seed(int64_t id, uint64_t* prn_seed, int offset)
```

Comment:
Since `prn_seed` is a pure output argument, we could change the interface to:
```C++
uint64_t init_seed(int64_t id, int offset)
```
This would help for the callers because they wouldn't have to declare a `uint64_t` without having a suitable initial value for it.